examples/update_flow_control/README.md

```diff
@@ -0,0 +1,178 @@
+{
+    "priority": -1,
+    "states": {
+        "ArtifactInstall_Enter": {
+            "action": "pause"
+        },
+        "ArtifactReboot_Enter": {
+            "action": "pause"
+        },
+        "ArtifactCommit_Enter": {
+            "action": "pause"
+        }
+    },
+    "id": "01234567-89ab-cdef-0123-456789abcdef"
+}
+-----------------------
+What do you want to do?
+  0) Pause on all states
+  1) Continue with Installing new software
+  2) Continue with Rebooting
+  3) Continue with Committing new software
+  q) Quit
+
+```
+
+## Implementation
+
+The example application implements two threads that communicate with two queues;
+one thread is responsible for managing the Mender D-Bus API communication, and
+another is responsible for dealing with the user input:
+
+* `do_handle_update_control_map`: initializes the D-Bus communication with the
+  Mender client, sets a "pause all" Update Control Map, and enters the main
+  event loop. From this loop two events might happen: that a new map is received
+  in the queue, or that the timeout from the current map is about to expire and
+  the map needs refreshing. On D-Bus errors, the error message is reported in
+  the corresponding queue and the thread exists.
+
+* `do_main_interactive`: print the selection menu to the user and ask for what
+  action to do next. When the user selects an action, the corresponding map is
+  sent to the que for the D-Bus control thread to handle. It constantly
+  refreshes the terminal to always display the current map on top of the user
+  options menu. If a D-Bus error is received in the corresponding queue, the
+  error message is printed to the user and the thread exists.
+
+## Going to production
+
+This example is meant to educate and show how the Mender D-Bus API works. There
+are few aspects that you need to take into consideration when writing a production
+grade application based on this example.
+
+### Selecting a UUID
+
+This example uses a dummy [Universally Unique Identifier, or
+UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier). Make sure to
+chose a new random UUID for your application that will stay constant.
+
+### Error handling and failure recovery
+
+The code in this example just prints the message and exists on any D-Bus error.
+A production application shall log the error but try to recover from it. Some of
+the actions that can be taken are retrying at a later time, or reporting the
+error to the user in a better way.
+
+# Refreshing of the maps
+
+The example constantly sets the "current" map. This is necessary for the Mender
+client to keep the map active.
+
+When a map expires the map is not taken into account anymore for a Mender update
+and by default the Mender client will continue on any step (assuming there no
+other maps). The application can however customize what happens when a map
+expires using its `on_map_expire` attribute in the map, which can be set to
+`continue`, `force_continue`, or `fail`.
```

Comment:
Should we write that the motivation for expiring maps is to have a last resort when a frontend process has crashed or is hanging? Not sure if it makes it clearer, or if I'm just too close to the design and want to tell everything. I leave it to you to decide, since you have a lower bias.

Reply:
It might be too much of a detail, but I'll add a sentence. Makes sense to my "lower bias" point of view.

Reply:
I rewrote this paragraph like:

```
When a map expires it is not taken into account anymore for a Mender update and
by default the Mender client will continue on any step (assuming there no other
maps). This is a safety mechanism for the device to always be able to be updated
when a frontend process has crashed or is hanging.
```

Reply:
:+1: